update interface name and version

funkymalc opened the pull request:
The interface name and interface version specified in the module descriptor should be distinct from the module name and version.    An interface can be independent of a module .  Also, I don't think you want to increment the interface version each time a new version of the module is released (snapshot or release version).  Probably should only be changed when the interface has changed.

sosguthorpe commented:
Actually I would disagree about the versions. When we introduce none breaking changes to the API (or interface) then this should map directly onto an increment of both the module's and interfaces minor versions. The interface version only tracked the `major.minor` version which I think is correct. That way, patches, hot-fixes, snaptshots etc do not increment the version here, but more major changes do.

I take your point as for the interface name though. Although we plan to maintain a 1-1 relationship between interface and module, I think "mod-something" isn't necessarily a good name for what if effectively an API.

Would be happy to accept the name change, but think the semantics of the version increment stand. Will ask @ianibbo to look this over too.

doytch commented:
In terms of naming, the general pattern for other modules and their main interface is just stripping off the `mod-`. So if we're `mod-erm`, `erm` might be cleaner and easier to find when grepping through the interfaces output. I'm under the impression that the `olf-` stuff is just a holdover from previous code, right? 

But 🤷‍♂️, it doesn't really affect me 😆

funkymalc commented:
@sosguthorpe - ok.  I've made a couple of updates to this PR:

1.  Interface name = 'erm' in the module descriptor
2. I take your point re: interface versions and reverted previous change.    Updated the version_regex in build.gradle to properly handle snapshot versions that have the format '-SNAPSHOT.$BUILD_NUMBER'
